plugins/org.python.pydev/src/org/python/pydev/editor/correctionassist/heuristics/AssistFString.java

```diff
@@ -0,0 +1,181 @@
+                variablesBuf.deleteFirst();
+                variablesBuf.deleteLast();
+                variablesBuf.trim();
+            }
+            if (variablesBuf.endsWith(',')) {
+                variablesBuf.deleteLast();
+                variablesBuf.trim();
+            }
+
+            // initialize format output string with the properly size allocation (f + string len + variables len)
+            FastStringBuffer strBuf = new FastStringBuffer(1 + partition.getLength() + variablesBuf.length());
+            strBuf.append('f').append(stringPartitionContents);
+
+            // get variables without literals
+            String variablesWithoutLiterals = PySelection.getLineWithoutCommentsOrLiterals(variablesBuf.toString());
+            // get variable by variable
+            List<String> variables = new ArrayList<String>();
+            parsingUtils = ParsingUtils.create(variablesBuf);
+            int j = 0;
+            while (j < parsingUtils.len()) {
+                char c = parsingUtils.charAt(j);
+                if (Character.isWhitespace(c) || c == ',') {
+                    j++;
+                    continue;
+                }
+                int commaPos;
+                if (c == '{' || c == '[' || c == '(' || c == '"' || c == '\'') {
+                    commaPos = parsingUtils.eatPar(j, null, c) + 1;
+                } else {
+                    commaPos = variablesWithoutLiterals.indexOf(',', j);
+                }
+                if (commaPos != -1) {
+                    CharSequence subSequence = variablesBuf.subSequence(j, commaPos);
+                    variables.add(subSequence.toString().trim());
+                    j = commaPos + 1;
+                } else {
+                    CharSequence subSequence = variablesBuf.subSequence(j, variablesBuf.length());
+                    variables.add(subSequence.toString().trim());
+                    break;
+                }
+            }
+            // iterate through variables and edit the f-string output
+            if (formatCount == variables.size()) {
+                for (String variable : variables) {
+                    strBuf.replaceFirst("%s", "{" + variable + "}");
+                }
+            } else if (variables.size() == 1) {
+                String variable = variables.get(0);
+                for (j = 0; j < formatCount; j++) {
+                    strBuf.replaceFirst("%s", "{" + variable + "[" + j + "]" + "}");
+                }
+            } else {
+                return lst;
+            }
+            lst.add(CompletionProposalFactory.get().createPyCompletionProposal(strBuf.toString(),
+                    partition.getOffset(), i - partition.getOffset(), 0, getImage(imageCache,
+                            UIConstants.COMPLETION_TEMPLATE),
+                    "Convert to f-string", null, null, IPyCompletionProposal.PRIORITY_DEFAULT, null));
+        } catch (SyntaxErrorException e) {
+            Log.log(e);
```

Comment:
I don't think we should log anything in this case (the code is not well-formed, so, we shouldn't complain).